### DISCUSSION TITLE
REDASH MULTI ORG Megathread. Is anyone using it?

### DISCUSSION THREAD

**Discussion by gpspake** [getredash/redash#6839] [General]
We're setting up Redash as a tool for customers and running in to some permission related challenges like users being able to see other user's queries and data sources. We've experimented with moving certain permissions like `list_data_sources` and `list_users` to the admin group but those changes come with their own problems like users not being able to see data sources in the query editor.

Multi org seems like it could be a viable solution for what we're trying to do here but it's not documented or officially supported; it sounds like it's left over from when Redash was offered as a hosted solution. Still there are some threads about it and it seems like it may still be usable with some manual configuration. Is anyone doing this?

Here are some discussions I've found. If there are any others please share:

- 2017 discussion from the archived/read-only redash forum  (last active 2/22) [Providing redash to our customers](https://discuss.redash.io/t/providing-redash-to-our-customers/1111) - per arikfr: _"Redash does have multi-tenancy support, this is what we use to power our hosted version (https://app.redash.io/). So you can have a single deployment of Redash, where you create an account for each of your users. Then you can use the API, to create the data sources, user accounts, and default visualizations. Multi-tenancy is not something we provide support for via the open channels, but you’re welcome to reach out by email "_

- 2018 discussion from the archived/read-only redash forum (last active 2/22): [Creating organizations](https://discuss.redash.io/t/creating-organizations/2540)  - Sounds like at least one user may have been able to run self hosted multi org successfully.

- 2021 discussion from the archived/read-only redash forum: "[Using Redash as Muti-Tenant application](https://discuss.redash.io/t/using-redash-as-muti-tenant-application/9357)" 

- 2023 discussion from GH: [multiorg](https://github.com/getredash/redash/discussions/6002) - not a lot of discussion but shows there's still interest.

- 2018 Medium post: [Redash as a self service multi-tenant BI solution](https://medium.com/@manoveg/redash-as-a-self-service-multi-tenant-bi-solution-b31edbfb5c6d) - it sounds like the host achieved some multi-tenant features without multi org, using groups. I might revisit this option.

**Comment by justinclift** [comment]
"Multi-org" seems like an approach that was used by the original Redash team for their online SaaS service, but hasn't really received any attention (or been kept in mind) with the relaunch as a Community Led project.

Personally speaking - and not being familiar with multi-org myself as I don't use it either - I wouldn't be deploying Redash in a multi-org configuration.  Instead I'd look at using individual Redash instances for each customer.

At least, not until it's been looked at properly by current developers and made sure it definitely works correctly.

**Comment by eradman** [comment]
I made several changes to make this feature accessible again

https://github.com/getredash/redash/pull/6919 
https://github.com/getredash/redash/pull/6923 
https://github.com/getredash/redash/pull/6927

Basic setup:

(1) Enable MULTI_ORG in `.env` 

```
REDASH_MULTI_ORG=true
```

(2) Update the slug for the default organization

```
$ psql -h localhost -p 15432 -U postgres
postgres=# UPDATE organizations SET slug ='redash' WHERE slug='default';
```

(3) Log in using http://localhost:5001/redash/

There are several use cases for this feature, so it is worth further testing.

**Reply by justinclift** [reply]
> further testing.

Definitely.  Multi-org seems designed to have lots of different users/customers in a single Redash installation.

It'd only take one (internal) Redash SQL query to be missing the right WHERE clause for things to go badly wrong and have customers seeing each others data. (!)

That worries me a bit.
